pulsefire/ratelimiters.py

```diff
@@ -53,15 +53,17 @@ class RiotAPIRateLimiter(BaseRateLimiter):
     Parameters:
         proxy: URL of the proxy rate limiter.
         proxy_secret: Secret of the proxy rate limiter if required.
+        limiting_share: Value from 0 to 1. Rate limiter will only allow requests up to `bucket_max * limiting_share`.
     """
 
     _index: dict[tuple[str, int, *tuple[str]], tuple[int, int, float, float, float]] = \
         collections.defaultdict(lambda: (0, 0, 0, 0, 0))
 
-    def __init__(self, *, proxy: str | None = None, proxy_secret: str | None = None) -> None:
+    def __init__(self, *, proxy: str | None = None, proxy_secret: str | None = None, limiting_share: float = 1) -> None:
         self.proxy = proxy
         self.proxy_secret = proxy_secret
         self._track_syncs: dict[str, tuple[float, list]] = {}
+        self.limiting_share = max(0, min(1, limiting_share))
```

Comment:
While this does restrict the coefficient to the range of [0, 1], it does so silently and we don't want that, please use an assert line with an error message: `assert 1 >= limit_perc >= 0, "limit_perc must be a value within the range [0, 1]"`